### PR TITLE
[Compression] Update sparsity calculation lifecycle when fetching the compressor

### PR DESCRIPTION
**Pull request by dsikka**
# Summary
- Updating the lifecycle based on feedback from research and meta
- #1293 is addressed @eldarkurtic 

### 1. No longer always infer sparsity 
- If there is existing sparsity in the model that should be accounted for while compressing the model, the user is required to indicate this by providing a sparsity config. This is not a common case and primarily something we've seen from research. As a result, it does not need to be the default pathway
- If no sparsity config is provided, only check for sparsity if a recipe with sparsity was applied OR the user explicitly would like to apply a sparsity compressor by setting `skip_compression_stats` parameter
- Based on these changes, we no longer use the global_sparsity to dictate whether or not a sparse compressor is applied

### 5. Fix log outputs to prevent cutting off

# Follow-ups:
- Need to update passing saving arguments properly - they are currently not being passed apart from save_compressed and there is no clear API on what the arguments are/how they should be passed, both to save_pretrained and when providing an output dir to oneshot

# Testing
- Tested a sparse-only model
- Tested a model with both sparsity and quantization
- Tested a model with existing sparsity
- Tested a model with only quantization

For all cases, the model compresses to the expected format.

### CODE REVIEW

File: src/llmcompressor/transformers/sparsification/compressed_tensors_utils.py [hunk 1]
```diff
@@ -58,7 +58,7 @@ def save_pretrained_wrapper(
             quantization_format: Optional[str] = None,
             save_compressed: bool = True,
             safe_serialization: bool = True,
-            skip_compression_stats: bool = False,
+            skip_sparsity_compression_stats: bool = False,
```
Comment (rahul-tuli) [inline src/llmcompressor/transformers/sparsification/compressed_tensors_utils.py:61]:
```suggestion
            skip_sparsity_compression_stats: bool = True,
```

Since it's True in get_model_compressor as well

**Review by rahul-tuli — COMMENTED** [review]
Nice, the functionality seems to be all there! LGTM pending a small nit and some clean up (docstrings and such)

**Review by brian-dellabetta — DISMISSED** [review]

File: tests/llmcompressor/transformers/sparsification/test_compress_tensor_utils.py [hunk 2]
```diff
@@ -537,7 +542,7 @@ def __init__(self, weights, weight_scale=None, weight_zero_point=None):
 
         # Linear layer without bias
         self.linear = nn.Linear(in_features, out_features, bias=False)
-        self.linear.weight = nn.Parameter(weights, requires_grad=False)
+        self.linear.weight = nn.Parameter(weights, requires_grad=True)
```
Comment (brian-dellabetta) [inline tests/llmcompressor/transformers/sparsification/test_compress_tensor_utils.py:545]: we need gradients here?

**Review by brian-dellabetta — DISMISSED** [review]

**Review by brian-dellabetta — APPROVED** [review]

File: tests/llmcompressor/transformers/sparsification/test_compress_tensor_utils.py (hunk 2 above)
Comment (dsikka) [reply tests/llmcompressor/transformers/sparsification/test_compress_tensor_utils.py:546]: When generating a sparsity config, we calculate the global_sparsity, which looks at trainable parameters. The test was skipping this step before so it did not matter previously.

File: src/llmcompressor/transformers/finetune/session_mixin.py [hunk 3]
```diff
@@ -358,7 +358,13 @@ def train(self, *args, stage: Optional[str] = None, **kwargs):
 
         return output
 
-    def save_model(self, output_dir: str, _internal_call: bool = False):
+    # TODO: support all save args, not just skip_sparsity_compression_stats
+    def save_model(
+        self,
+        output_dir: str,
+        _internal_call: bool = False,
+        skip_sparsity_compression_stats: Optional[bool] = False,
```
[the remark this answers is not in the document]
Comment (dsikka) [reply src/llmcompressor/transformers/finetune/session_mixin.py:366]: We don’t support passing in saving args to the train pathway right now. I will enable this in a follow up. So we need this param to be set to True while training. However for one shot, we support this. Hence different defaults. Will be consistent when I had the saving params

**Review by brian-dellabetta — APPROVED** [review]

**Review by rahul-tuli — APPROVED** [review]

**Review by kylesayrs — APPROVED** [review]
